modules/rostests/apitests/ntdll/RtlAllocateHeap.c

```diff
@@ -9,58 +9,62 @@
 
 #include <pseh/pseh2.h>
 
-PVOID Buffers[0x100];
-
 START_TEST(RtlAllocateHeap)
 {
+    PVOID Buffers[0x100];
     USHORT i;
     HANDLE hHeap;
-    BOOLEAN Aligned = TRUE;
+    BOOLEAN Not8BytesAligned = FALSE;
+    BOOLEAN Not16BytesAligned = FALSE;
     RTL_HEAP_PARAMETERS Parameters = {0};
 
-    for (i = 0; i < 0x100; ++i)
+    for (i = 0; i < ARRAYSIZE(Buffers); ++i)
     {
         Buffers[i] = RtlAllocateHeap(RtlGetProcessHeap(), HEAP_CREATE_ALIGN_16, (i % 16 ) + 1);
         ASSERT(Buffers[i] != NULL);
-        if (!((ULONG_PTR)Buffers[i] & 0xF))
-        {
-            Aligned = FALSE;
-        }
+
+        if (Buffers[i] != ALIGN_DOWN_POINTER_BY(Buffers[i], 8))
+            Not8BytesAligned = TRUE;
+
+        if (Buffers[i] != ALIGN_DOWN_POINTER_BY(Buffers[i], 16))
+            Not16BytesAligned = TRUE;
     }
 
-    for (i = 0; i < 0x100; ++i)
+    for (i = 0; i < ARRAYSIZE(Buffers); ++i)
     {
         RtlFreeHeap(RtlGetProcessHeap(), 0, Buffers[i]);
     }
 
-    ok(Aligned == FALSE, "No unaligned address returned\n");
+    ok(Not8BytesAligned == FALSE, "Found address that was not 8 byte aligned\n");
+#ifdef _WIN64
+    ok(Not16BytesAligned == FALSE, "Found address that was not 16 byte aligned\n");
+#endif
 
-    Aligned = TRUE;
+    Not16BytesAligned = FALSE;
     Parameters.Length = sizeof(Parameters);
     hHeap = RtlCreateHeap(HEAP_CREATE_ALIGN_16, NULL, 0, 0, NULL, &Parameters);
     if (hHeap == NULL)
     {
         return;
     }
 
-    for (i = 0; i < 0x100; ++i)
+    for (i = 0; i < ARRAYSIZE(Buffers); ++i)
     {
         Buffers[i] = RtlAllocateHeap(hHeap, 0, (i % 16 ) + 1);
         ASSERT(Buffers[i] != NULL);
```

Comment:
Why `ASSERT()`, instead of "handling" the failure?

Reply:
There is no 'handling' this, just like `RtlCreateHeap` failing.
Out of memory is gameover.